### PR TITLE
DOCS-1910: Include new model frameworks beside TFLite 

### PR DESCRIPTION
**Pull request by sguequierre**
Scope of this ticket was a little tricky but I think this is everything that's not covered [here](https://github.com/viamrobotics/docs/pull/2628)

* Confirmed with @sbal13 new frameworks should work with vision service

### CODE REVIEW

**Comment by npentrel** [comment]
@sguequierre I wonder if the model types are now available for `get_modular_resources.py` to ingest and for us to display in the Ml models table...

**Review by npentrel — CHANGES_REQUESTED** [review]
Please add the information from the file included on the ticket about what the different frameworks are

**Comment by sguequierre** [comment]
> Please add the information from the file included on the ticket about what the different frameworks are

Sorry, what do you mean? From here https://docs.google.com/document/d/1IBL2H2IL618YZuR_FSAe6FzA5_hbmqQ-fmrd_xBHLOg/edit#heading=h.slcmrvcghves? I have the "model frameworks" column

**Comment by sguequierre** [comment]
> @sguequierre I wonder if the model types are now available for `get_modular_resources.py` to ingest and for us to display in the Ml models table...

from a quick investigation I think possibly? No specific field but looking at https://python.viam.dev/autoapi/viam/gen/app/v1/app_pb2/index.html#viam.gen.app.v1.app_pb2.RegistryItem it might be available in ModuleMetadata. I can't tell from that object's docs though. Made a ticket to figure it out https://viam.atlassian.net/browse/DOCS-1970

File: docs/ml/upload-model.md [hunk 1]
```diff
@@ -14,12 +14,23 @@ description: "Upload a Machine Learning model to Viam to use it with the ML Mode
...
 Also, [share a model from your organization](#make-an-existing-model-public-in-the-registry) with other users in the registry.
 
+## Model framework support
+
+Before uploading your model to the cloud, check and see if you will be able to deploy it to your machine.
+Model framework support is currently as follows:
+
+| Model Framework | Hardware Support | System Architecture | Description |
+| --------------- | ---------------- | ------------------- | ----------- |
+| [TensorFlow Lite](https://www.tensorflow.org/lite) | Any CPU <br> Nvidia GPU | Linux, Raspbian | Quantized version of TensorFlow that has reduced compatibility for models but supports more hardware. |
```
Comment (sguequierre) [inline docs/ml/upload-model.md:29]: not sure if System Architecture column is accurate: asked question [here](https://docs.google.com/document/d/1IBL2H2IL618YZuR_FSAe6FzA5_hbmqQ-fmrd_xBHLOg/edit?disco=AAABJEf7MMI)

Comment (sguequierre) [reply docs/ml/upload-model.md:30]: @npentrel I also made a ticket here https://viam.atlassian.net/browse/DOCS-1980 -- table is not fully complete as support is not complete

Comment (npentrel) [reply docs/ml/upload-model.md:30]: OK - maybe we comment out the ones that aren't supported yet and then uncomment once supported?

File: docs/ml/upload-model.md [hunk 2]
```diff
@@ -14,12 +14,24 @@ description: "Upload a Machine Learning model to Viam to use it with the ML Mode
 The ML model service works with models trained inside and outside the Viam app.
 To use a model that you have trained yourself outside the Viam app, [upload it as a new model privately or share it in the Viam registry](#upload-a-new-model-or-new-version).
 If you need to update a previously uploaded model, you can also [upload a new version](#upload-a-new-model-or-new-version).
-To work with the `tflite_cpu` ML model service, an ML model is comprised of a <file>.tflite</file> model file which defines the model, and optionally a <file>.txt</file> labels file which provides the text labels for your model.
```
Comment (npentrel) [inline docs/ml/upload-model.md:17]: Should we add this info to the table or somewhere add info for each of the frameworks?

Comment (sguequierre) [reply docs/ml/upload-model.md:17]: Adding that + all the info steven said was needed to the upload model page in this pr https://github.com/viamrobotics/docs/pull/2628/files

File: docs/ml/upload-model.md [hunk 3]
```diff
@@ -14,12 +14,24 @@ description: "Upload a Machine Learning model to Viam to use it with the ML Mode
...
 
+## Model framework support
+
+Before uploading your model to the cloud, check and see if you will be able to deploy it to your machine.
+Model framework support is currently as follows:
+
+<!-- prettier-ignore -->
+| Model Framework | Hardware Support | System Architecture | Description |
+| --------------- | ---------------- | ------------------- | ----------- |
+| [TensorFlow Lite](https://www.tensorflow.org/lite) | Any CPU <br> Nvidia GPU | Linux, Raspbian | Quantized version of TensorFlow that has reduced compatibility for models but supports more hardware. |
```
Comment (sguequierre) [inline docs/ml/upload-model.md:30]:
```suggestion
| [TensorFlow Lite](https://www.tensorflow.org/lite) | Any CPU <br> Nvidia GPU | Linux, Raspbian, MacOS, Android | Quantized version of TensorFlow that has reduced compatibility for models but supports more hardware. |
```

**Review by npentrel — APPROVED** [review]
Good work - good  catches that you're updating all these references!

File: docs/ml/upload-model.md (hunk 1 above)
Comment (sguequierre) [reply docs/ml/upload-model.md]: I think as they're all supported (just support to be extended further for PyTorch and TensorFlow) I will keep but update?

**Pull request merged**